Ably-js v2 migration

VeskeR opened the pull request:
Updates this project to use ably-js v2 so it matches updated code samples in related blog post: https://ably.com/blog/next-js-vercel-link-sharing-serverless-websockets

VeskeR committed:
npm uninstall @ably-labs/react-hooks

VeskeR committed:
npm install --save ably@2

VeskeR committed:
Formatting

VeskeR committed:
Migrate to ably-js v2

VeskeR committed:
Update README

VeskeR committed:
Fix "useLayoutEffect does nothing on the server" warnings

ttypic commented on components/Articles.js:
```diff
@@ -59,18 +61,18 @@ export default function Articles(props) {
       <form onSubmit={handleFormSubmission} className={styles.form}>
         <input
-          type="text"
+          type='text'
```
small note: a bit odd codestyle, I believe double quotes more common in jsx

VeskeR replied:
Yeah, my VS Code automatically replaced it with single quotes, not sure why it did that for html attributes as it doesn't do it for the regular js code (see import statements for example, all of them remained in double quotes as it was before in this repo).
However, all of this is coming from the lack of prettier config in this project. I'll add a prettier config file and run format script in a separate PR, and then will rebase this PR on top of the formatting one to keep meanigful changes clear for the review

ttypic commented on yarn.lock:
Not sure it's good to keep both `yarn.lock` and `package-lock.json`

VeskeR replied:
Yeah, I was a bit confused why project had both of them too, and wasn't sure if I wanted to delete one of them. Since you also don't like it, I will remove `yarn.lock` in preference of `package-lock.json` as I am mainly working with npm.

ttypic reviewed (approved):
Looks good to me, added couple of minor suggestions

VeskeR committed:
Delete `yarn.lock`

No need to use both `yarn.lock` and `package-lock.json` and we prefer
to have `package-lock.json`.

VeskeR commented:
@ttypic Removed `yarn.lock` in this PR and updated formatting in https://github.com/ably-labs/ably-next-vercel-news/pull/18 PR. This PR is now based on that formatting PR so please review that one too, thank you!

VeskeR committed:
Update broken link in CONTRIBUTING

Previous link to the guide for resolving merge conflicts in github no
longer exist, so replaced with the next best tutorial from github docs.
